### PR TITLE
JSONSchema processor

### PR DESCRIPTION
**Pull request by stephanebruckert**
This is to solve https://github.com/Jeffail/benthos/issues/313

Most of the processor comes from the JSONSchema condition and was adapted:
- https://github.com/Jeffail/benthos/blob/master/lib/condition/jsonschema.go
- https://github.com/Jeffail/benthos/blob/master/lib/condition/jsonschema_test.go

However I am not sure how to test that the ${!error} will contain the message I need? Other processor tests don't seem to ever test the error field.

Thx for your feedback!

### CODE REVIEW

File: lib/processor/constructor.go [hunk 1]
```diff
@@ -163,6 +164,7 @@ type Config struct {
 	InsertPart   InsertPartConfig   `json:"insert_part" yaml:"insert_part"`
 	JMESPath     JMESPathConfig     `json:"jmespath" yaml:"jmespath"`
 	JSON         JSONConfig         `json:"json" yaml:"json"`
+	JSONSchema   JSONSchemaConfig   `json:"jsonschema" yaml:"jsonschema"`
```
Comment (Jeffail) [inline lib/processor/constructor.go]: The tags here need to match the type name `json_schema`

**Comment by Jeffail** [comment]
Hey @stephanebruckert, error messages are stored as a metadata field, so to check the contents of the failure flag you can use `part.Metadata().Get(FailFlagKey)`, where `FailFlagKey` is the static string `benthos_processing_failed`.

**Comment by stephanebruckert** [comment]
Hey @Jeffail, thanks, that helped! PR is now ready for full review

**Comment by Jeffail** [comment]
Awesome, thanks @stephanebruckert!

**Comment by stephanebruckert** [comment]
@Jeffail thanks a lot for merging this. Going to use it from today! 

However even though a tag hasn't been created yet, should it not have automatically updated the `latest` docker image? https://hub.docker.com/r/jeffail/benthos/tags

**Comment by Jeffail** [comment]
@stephanebruckert, the latest tag only gets updated with a new tag or release candidate. I'm happy to put a build out today though, it'll be v3.5.0.

**Pull request merged**